test/Driver/lines_to_execute.swift

```diff
@@ -0,0 +1,26 @@
+// RUN: %swift_driver_plain -e 'print("Hello, world!")' 2>&1 | %FileCheck -check-prefix CHECK-1 %s
+// CHECK-1: Hello, world!
+
+// RUN: not %swift_driver_plain -e 'foo' 2>&1 | %FileCheck -check-prefix CHECK-2 %s
+// CHECK-2: -e:1:1: error: use of unresolved identifier 'foo'
+
+// RUN: %swift_driver_plain -e 'print("Hello, world!")' -e 'print("Hi, land!")' 2>&1 | %FileCheck -check-prefix CHECK-3 %s
+// CHECK-3: Hello, world!
+// CHECK-3: Hi, land!
+
+// RUN: not %swift_driver_plain -e 'print("Hello, world!")' -e 'foo' 2>&1 | %FileCheck -check-prefix CHECK-4 %s
+// CHECK-4: -e:2:1: error: use of unresolved identifier 'foo'
+// CHECK-4-NOT: Hello, world!
+
+// RUN: %swift_driver_plain -MSwiftShims -e '_stdlib_write(1, "Hello, world!", 13)' 2>&1 | %FileCheck -check-prefix CHECK-5 %s
+// CHECK-5: Hello, world!
+
+// RUN: not %swift_driver_plain -MFnord -e 'fnord' 2>&1 | %FileCheck -check-prefix CHECK-6 %s
+// CHECK-6: -M:1:8: error: no such module 'Fnord'
+// CHECK-6-NOT: -e:1:1: error: use of unresolved identifier 'fnord'
+
+// RUN: not %swift_driver_plain -MSwiftShims -e 'fnord' 2>&1 | %FileCheck -check-prefix CHECK-7 %s
+// CHECK-7: -e:1:1: error: use of unresolved identifier 'fnord'
+
+// RUN: not %swift_driver_plain -MSwiftShims 2>&1 | %FileCheck -check-prefix CHECK-8 %s
+// CHECK-8: error: cannot use -M arguments to import modules without -e arguments to use them
```

Comment:
Can you check that `-M` and `-e` are rejected with `%swiftc_driver`, and that you can't mix `-e` and regular files?

Reply:
`-e` happens to *usually* not work with other files (it causes "expressions are not allowed at the top level" errors), but I'll add a specific, reliable diagnostic for that case.

I'd like `-e` to work with at least the frontend modes you might use when debugging how the compiler handles small code samples (like `-dump-ast`, `-emit-sil`, etc.) Just glancing at the `ActionType` enum, here's what I think we should support:

* Parse
* ResolveImports
* Typecheck
* DumpParse
* EmitSyntax
* DumpAST
* PrintAST
* DumpScopeMaps
* DumpTypeRefinementContexts
* EmitImportedModules
* EmitPCH
* EmitSILGen
* EmitSIL
* EmitSIBGen
* EmitSIB
* Immediate
* EmitAssembly
* EmitIR
* EmitBC
* DumpTypeInfo

And here's what we should prevent:

* NoneAction
* DumpInterfaceHash
* EmitModuleOnly
* MergeModules
* REPL (it might actually be a good idea to allow use with the REPL to set up an environment, but that'll require additional work, and we'd want to support files for this purpose at the same time)
* EmitObject

But this is just off the top of my head, and I don't know much about a few of these modes (like DumpInterfaceHash). Is there anything you'd change about these lists, or do you think this is totally the wrong approach?